.github/workflows/ci.yml

```diff
@@ -0,0 +1,137 @@
+      # appropriate here.
+      # FIXME(#9634): remove this once the issue is resolved.
+      - run: sudo sysctl vm.overcommit_memory=1 || true
+      - uses: actions/checkout@v2
+      - uses: baptiste0928/cargo-install@21a18ba3bf4a184d1804e8b759930d3471b1c941
+        with:
+          crate: cargo-nextest
+      - uses: baptiste0928/cargo-install@21a18ba3bf4a184d1804e8b759930d3471b1c941
+        with:
+          crate: cargo-deny
+      - uses: Swatinem/rust-cache@a95ba195448af2da9b00fb742d14ffaaf3c21f43
+        with:
+          prefix-key: "0" # change this to invalidate CI cache
+      - run: cargo nextest run --locked --workspace -p '*' --cargo-profile quick-release --profile ci ${{ matrix.flags }}
+        env:
+          RUST_BACKTRACE: short
+
+  db_migration:
+    name: "Database Migration"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: run tests/sanity/db_migration.py
+        run: |
+          cd pytest
+          pip3 install --user -r requirements.txt
+          python3 tests/sanity/db_migration.py
+
+  protobuf_backward_compat:
+    name: "Protobuf Backward Compatibility"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: bufbuild/buf-setup-action@v1
+      - uses: bufbuild/buf-breaking-action@v1
+        with:
+          against: "https://github.com/near/nearcore.git#ref=${{ github.event.pull_request.base.sha || github.event.merge_group.base_sha }}"
+
+  sanity_checks:
+    name: "Sanity Checks"
+    runs-on: ubuntu-22.04-16core
+    strategy:
+      fail-fast: false
+    timeout-minutes: 90
+
+    steps:
+      - uses: actions/checkout@v2
+      - run: cargo build -p neard --bin neard --features nightly
+      - name: run spin_up_cluster.py
+        # Note: We're not running spin_up_cluster.py for non-nightly
+        # because spinning up non-nightly clusters is already covered
+        # by other steps in the CI, e.g. upgradable.
+        run: |
+          cd pytest
+          python3 -m pip install --progress-bar off --user -r requirements.txt
+          python3 tests/sanity/spin_up_cluster.py
+
+      - run: cargo build -p neard --bin neard
+      - run: python3 scripts/state/update_res.py check
+
+      - run: python3 scripts/check_nightly.py
+      - run: python3 scripts/check_pytests.py
+      - run: python3 scripts/check_fuzzing.py
+
+      - run: python3 scripts/fix_nightly_feature_flags.py
+
+      - run: ./scripts/formatting --check
+
+      - name: check rpc_errors_schema.json
+        run: |
+          rm -f target/rpc_errors_schema.json
+          cargo check -p near-jsonrpc --features dump_errors_schema
+          if ! git --no-pager diff --no-index chain/jsonrpc/res/rpc_errors_schema.json target/rpc_errors_schema.json; then
+              set +x
+              echo 'The RPC errors schema reflects outdated typing structure; please run'
+              echo '    ./chain/jsonrpc/build_errors_schema.sh'
```

Comment:
I wonder why it is that we are checking one command, and suggesting users to use a different one. Sounds to me like we could run `./chain/jsonrpc/build_errors_schema.sh` in the CI and use the more `git diff --check` to verify no changes to checked-in files have occurred after the file got regenerated…?

Reply:
Literally no idea, it’s also why I tried to copy-paste this step as close as possible to what was in buildkite; I don’t really understand it all and the different interactions between its components 😅 but what you’re saying definitely makes sense!

Reply:
Can you transplant these comments into the workflow file as TODOs? It’ll be easier to check them later this way.